tests/drivers/counter/counter_basic_api/src/test_counter.c

```diff
@@ -157,19 +170,15 @@ void test_single_shot_alarm_instance(const char *dev_name)
 	err = counter_set_ch_alarm(dev, &alarm_cfg);
 	zassert_equal(0, err, "Counter set alarm failed\n");
 
-	k_busy_wait(1.2*counter_ticks_to_us(dev, ticks));
-
+	k_busy_wait(1.5*ALARM_US);
 	zassert_equal(1, alarm_cnt, "Expecting alarm callback\n");
 
-	k_busy_wait(counter_ticks_to_us(dev, 2*ticks));
+	k_busy_wait(1.5*ALARM_US);
 	zassert_equal(1, alarm_cnt, "Expecting alarm callback\n");
 
 	err = counter_disable_ch_alarm(dev, &alarm_cfg);
 	zassert_equal(0, err, "Counter disabling alarm failed\n");
 
-	err = counter_set_wrap(dev, counter_get_max_wrap(dev), NULL, NULL);
```

Comment:
you shouldn't remove that but rather accept -ENOTSUP error. It resets counter to default state. Once this is removed, on some platforms following tests may fail.

Reply:
maybe it would be good to split this test to part which does not require `set_wrap` (not `set_top_value`) and one that requires. Because now, you kind of limiting this test for platforms that support `set_wrap`.